pkg/resources/schema.go

```diff
@@ -29,268 +33,380 @@ var schemaSchema = map[string]*schema.Schema{
 		var err error
-		if managed.(bool) {
+		if _, ok := d.GetOk("with_managed_access"); ok {
 			err = client.Schemas.Alter(ctx, id, &sdk.AlterSchemaOptions{
-				EnableManagedAccess: sdk.Bool(true),
+				EnableManagedAccess: sdk.Pointer(true),
 			})
 		} else {
 			err = client.Schemas.Alter(ctx, id, &sdk.AlterSchemaOptions{
-				DisableManagedAccess: sdk.Bool(true),
+				DisableManagedAccess: sdk.Pointer(true),
 			})
 		}
 		if err != nil {
-			return fmt.Errorf("error changing management state on %v err = %w", d.Id(), err)
+			return diag.FromErr(fmt.Errorf("error changing management state on %v err = %w", d.Id(), err))
 		}
 	}
 
-	if d.HasChange("data_retention_days") {
-		if days := d.Get("data_retention_days"); days.(int) != IntDefault {
-			err := client.Schemas.Alter(ctx, id, &sdk.AlterSchemaOptions{
-				Set: &sdk.SchemaSet{
-					DataRetentionTimeInDays: sdk.Int(days.(int)),
-				},
-			})
-			if err != nil {
-				return fmt.Errorf("error setting data retention days on %v err = %w", d.Id(), err)
-			}
+	set := new(sdk.SchemaSet)
+	unset := new(sdk.SchemaUnset)
+
+	if d.HasChange("comment") {
+		comment := d.Get("comment").(string)
+		if len(comment) > 0 {
+			set.Comment = &comment
 		} else {
-			err := client.Schemas.Alter(ctx, id, &sdk.AlterSchemaOptions{
-				Unset: &sdk.SchemaUnset{
-					DataRetentionTimeInDays: sdk.Bool(true),
-				},
-			})
-			if err != nil {
-				return fmt.Errorf("error unsetting data retention days on %v err = %w", d.Id(), err)
-			}
+			unset.Comment = sdk.Bool(true)
 		}
 	}
 
-	if d.HasChange("tag") {
-		unsetTags, setTags := GetTagsDiff(d, "tag")
-
-		if len(unsetTags) > 0 {
-			err := client.Schemas.Alter(ctx, id, &sdk.AlterSchemaOptions{
-				UnsetTag: unsetTags,
-			})
-			if err != nil {
-				return fmt.Errorf("error occurred when dropping tags on %v, err = %w", d.Id(), err)
-			}
+	if updateParamDiags := HandleSchemaParametersChanges(d, set, unset); len(updateParamDiags) > 0 {
+		return updateParamDiags
+	}
+	if (*set != sdk.SchemaSet{}) {
+		err := client.Schemas.Alter(ctx, id, &sdk.AlterSchemaOptions{
+			Set: set,
+		})
+		if err != nil {
+			return diag.FromErr(err)
 		}
+	}
 
-		if len(setTags) > 0 {
-			err := client.Schemas.Alter(ctx, id, &sdk.AlterSchemaOptions{
-				SetTag: setTags,
-			})
-			if err != nil {
-				return fmt.Errorf("error occurred when setting tags on %v, err = %w", d.Id(), err)
-			}
+	if (*unset != sdk.SchemaUnset{}) {
+		err := client.Schemas.Alter(ctx, id, &sdk.AlterSchemaOptions{
+			Unset: unset,
+		})
+		if err != nil {
+			return diag.FromErr(err)
 		}
 	}
 
-	return ReadSchema(d, meta)
+	return ReadContextSchema(ctx, d, meta)
 }
 
-// DeleteSchema implements schema.DeleteFunc.
-func DeleteSchema(d *schema.ResourceData, meta interface{}) error {
-	client := meta.(*provider.Context).Client
-	ctx := context.Background()
+func HandleSchemaParametersChanges(d *schema.ResourceData, set *sdk.SchemaSet, unset *sdk.SchemaUnset) diag.Diagnostics {
```

Comment:
You were reusing stuff from the database, maybe the handling method could also be reused here?

Reply:
Well, this function uses different set/unset structs, so this would require extracting common parameters in general. I think this requires a separate discussion on our approach here.